base_layer/wallet/src/transaction_service/service.rs

```diff
@@ -642,9 +661,11 @@ where
             return Ok(tx_id);
         }
 
+        // TODO: Complete wiring of sending transactions with other tari script feature upstream
```

Comment:
The reason I wouldn't reuse this function is you are going to have two completely different behaviours in this function now. After you have called prepare_transaction_to_send you will handle this tx differently to this standard tx. Reusing this function means you are going to make this function huge with lots of sub branches. 

The pay to self above is just a special case of a traditional MW transaction where one-sided is a completely different beast.